File: platform/packages_py/env_resolver/env_resolver/env_anthropic_x.py
from dataclasses import dataclass
from typing import Any

from env_resolve import resolve


@dataclass(frozen=True)
class AnthropicEnv:
    api_key: str | None
    model: str


def resolve_anthropic_env(config: dict[str, Any] | None = None) -> AnthropicEnv:
    return AnthropicEnv(
        api_key=resolve(None, ['ANTHROPIC_API_KEY'], config, 'api_key', None),
        model=resolve(None, ['ANTHROPIC_MODEL'], config, 'model', 'claude-sonnet-4-5-20250514'),
    )

File: platform/packages_mjs/env-resolver/src/env-anthropic.x.ts
import { resolve } from '@internal/env-resolve';

export interface AnthropicEnv {
    apiKey: string | undefined;
    model: string;
}

export function resolveAnthropicEnv(config?: Record<string, any>): AnthropicEnv {
    return {
        apiKey: resolve(undefined, ['ANTHROPIC_API_KEY'], config, 'apiKey', undefined),
        model: resolve(undefined, ['ANTHROPIC_MODEL'], config, 'model', 'claude-sonnet-4-5-20250514'),
    };
}
